synapse/handlers/profile.py

```diff
@@ -19,23 +19,64 @@
 
 import synapse.types
 from synapse.api.errors import SynapseError, AuthError, CodeMessageException
-from synapse.types import UserID
+from synapse.types import UserID, get_domain_from_id
 from ._base import BaseHandler
 
-
 logger = logging.getLogger(__name__)
 
 
-class ProfileHandler(BaseHandler):
+class ProfileHandler(object):
+    PROFILE_UPDATE_MS = 60 * 1000
+    PROFILE_UPDATE_EVERY_MS = 24 * 60 * 60 * 1000
 
     def __init__(self, hs):
-        super(ProfileHandler, self).__init__(hs)
+        self.hs = hs
+        self.store = hs.get_datastore()
+        self.clock = hs.get_clock()
+        self.ratelimiter = hs.get_ratelimiter()
+
+        # AWFUL hack to get at BaseHandler.ratelimit
+        self.base_handler = BaseHandler(hs)
```

Comment:
I'm not sure I want to know why this is needed, but future generations deserve to know why we did what we did.

Reply:
I think I actually I can still just inherit from BaseHandler. At this point I think I was close to purging  all the unit tests with fire so may not have been at my best.....